File: MarketWatch-Backend/HomePage/src/main/java/com/team7/Miscellaneous/HomePage/model/Stock.java
package com.team7.Miscellaneous.HomePage.model;

import lombok.*;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class Stock {
    String id;
    String name;
    double instPrice;
    float percChange;
    long totalVolume;
    long marketCap;
    Description description;
    StockReport todaysReport;
    StockReport weeklyReport;
    StockReport monthlyReport;
    ESGScores esg;
    float esgScore;
    Chart chart;
}

File: MarketWatch-Frontend/src/app/models/stock-models/Stock.ts
import { Chart } from "chart.js";
import { MyChart } from "../chart-model/MyChart";
import { ESGScores } from "./ESGScores";
import { StockReport } from "./StockReport";

export class Stock{
    ide!: string;
    name!: string;
    instPrice!: number;
    percChange!: number;
    totalVolume!: number;
    marketCap!: number;
    description!: any;
    todaysReport!: StockReport;
    weeklyReport!: StockReport;
    monthlyReport!: StockReport;
    esg!: ESGScores;
    esgScore!: number;
    chart!:MyChart;
    color!:string;
    // constructor(){
    //     this.chart.lineChartData=[{
    //         data:this.todaysReport.history, label:'Todays Price'
    //     }];
    //     this.chart.lineChartLabels=[];
    //     this.chart.lineChartOptions={responsive:true};
    //     this.chart.lineChartColors=[{
    //         borderColor: 'black',
    //         backgroundColor: 'rgba(255,255,0,0.28)'
    //     }];
    //     this.chart.lineChartPlugins=[];
    //     this.chart.lineChartType='line';
    // } 
}